packages/compass-aggregations/src/components/pipeline-explain/explain-results.tsx

```diff
@@ -12,30 +12,45 @@ type ExplainResultsProps = {
 };
 
 const containerStyles = css({
-  display: 'flex',
-  flexDirection: 'column',
   gap: spacing[4],
+  display: 'grid',
+  gridTemplateRows: 'auto 1fr',
+  height: '100%',
+});
+
+const summaryStyles = css({
+  gridArea: 'summary',
 });
 
 const cardStyles = css({
-  // 170px works with minimum-height of compass
-  // todo: handle height for bigger sized compass
-  height: '170px',
+  gridArea: 'card',
+  maxHeight: '50vh',
   overflowY: 'scroll',
 });
 
 export const ExplainResults: React.FunctionComponent<ExplainResultsProps> = ({
   plan,
   stats,
 }) => {
+  const gridAreaStyles = css({
+    gridTemplateAreas: `
+      ${stats ? '"summary"' : ''}
```

Comment:
In the past I've tended to declare all styles at the top of the file then conditionally applied styles within the `className` using `cx`.   In this case it would be something like 
```ts
   <div
      className={cx(stats ? gridAreaWithStatsStyles : gridAreaWithoutStatsStyles, containerStyles)}
      data-testid="pipeline-explain-results"
    >
```

How do you feel about that as a practice? The goal is to keep the render functions smaller/more readable. cc @gribnoysup any preference?

Reply:
What you're suggesting makes sense, in addition to that `css` calls are not cheap so better to avoid this if possible (there are a few places already where we do this, but can clean it up, maybe a good idea for a custom warning rule)